From ef38b5768675c83cf070a5aebb783b5b129db60f Mon Sep 17 00:00:00 2001
From: Xie Chao <chao.xie@intel.com>
Date: Fri, 9 Aug 2019 15:44:58 +0800
Subject: [PATCH 2/3] UPSTREAM: mesa: android: Remove unnecessary dependency
 tracking rules
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

    The current AOSP master build system breaks building mesa due to the
    following error:

    external/mesa3d/src/compiler/Android.glsl.gen.mk:94: error:
      writing to readonly directory: "external/mesa3d/src/compiler/glsl/ir.h"

    This error is bogus -- nothing "writes" to ir.h -- but the rule is
    unnecessary because the generated header that is a dependency of the
    non-generated header should be added to LOCAL_GENERATED_SOURCES and this
    will track if the dependency needs to be regenerated.

    (This change fixes a similar problem affecting nir.h too.)

    Cc: Rob Clark <robdclark@chromium.org>
    Cc: Emil Velikov <emil.l.velikov@gmail.com>
    Cc: Amit Pundir <amit.pundir@linaro.org>
    Cc: Sumit Semwal <sumit.semwal@linaro.org>
    Cc: Alistair Strachan <astrachan@google.com>
    Cc: Greg Hartman <ghartman@google.com>
    Cc: Tapani Pälli <tapani.palli@intel.com>
    Cc: Jason Ekstrand <jason@jlekstrand.net>
    Reviewed-by: Tapani Pälli <tapani.palli@intel.com>
    Reviewed-by: Emil Velikov <emil.velikov@collabora.com>
    Signed-off-by: Alistair Strachan <astrachan@google.com>
    [jstultz: Forward ported and tweaked commit subject]
    Signed-off-by: John Stultz <john.stultz@linaro.org>
    (cherry picked from commit 0fda3eac3164dabdd85292e9a64d269b4fee3f4b)
    Tracked-On:OAM-82958
    Change-Id: Ia25250193f7dfbd3f353b3b927d9d09b437a38f9
    Reviewed-on: https://android.intel.com:443/672759

Change-Id: I1a9f3587446e3b4a389e11ad60cd4857fb77b54e
Tracked-On:
Signed-off-by: Xie Chao <chao.xie@intel.com>
Reviewed-on: 677880
Reviewed-by: Shi, Zhongjie <zhongjie.shi@intel.com>
Tested-by: Shi, Zhongjie <zhongjie.shi@intel.com>
---
 src/compiler/Android.glsl.gen.mk | 2 --
 src/compiler/Android.nir.gen.mk  | 2 --
 2 files changed, 4 deletions(-)

diff --git a/src/compiler/Android.glsl.gen.mk b/src/compiler/Android.glsl.gen.mk
index cdd786aa87..d73bb06e1c 100644
--- a/src/compiler/Android.glsl.gen.mk
+++ b/src/compiler/Android.glsl.gen.mk
@@ -90,8 +90,6 @@ $(intermediates)/glsl/glcpp/glcpp-lex.c: $(LOCAL_PATH)/glsl/glcpp/glcpp-lex.l
 $(intermediates)/glsl/glcpp/glcpp-parse.c: $(LOCAL_PATH)/glsl/glcpp/glcpp-parse.y
 	$(call glsl_local-y-to-c-and-h)
 
-$(LOCAL_PATH)/glsl/ir.h: $(intermediates)/glsl/ir_expression_operation.h
-
 $(intermediates)/glsl/ir_expression_operation.h: $(LOCAL_PATH)/glsl/ir_expression_operation.py
 	@mkdir -p $(dir $@)
 	$(hide) $(MESA_PYTHON2) $< enum > $@
diff --git a/src/compiler/Android.nir.gen.mk b/src/compiler/Android.nir.gen.mk
index 894fb12c4b..26115f446a 100644
--- a/src/compiler/Android.nir.gen.mk
+++ b/src/compiler/Android.nir.gen.mk
@@ -76,8 +76,6 @@ $(intermediates)/nir/nir_opcodes.h: $(nir_opcodes_h_deps)
 	@mkdir -p $(dir $@)
 	$(hide) $(MESA_PYTHON2) $(nir_opcodes_h_gen) $< > $@
 
-$(LOCAL_PATH)/nir/nir.h: $(intermediates)/nir/nir_opcodes.h
-
 nir_opcodes_c_gen := $(LOCAL_PATH)/nir/nir_opcodes_c.py
 nir_opcodes_c_deps := \
 	$(LOCAL_PATH)/nir/nir_opcodes.py \
-- 
2.21.0

